### DISCUSSION TITLE
Hyper 1.0 client side upgrade doesn't work for HTTP CONNECT

### DISCUSSION THREAD

**Discussion by OldMansKid** [hyperium/hyper#3502] [Q&A] [answered]
Hello hyper maintainers, I'm trying to let my Hyper client support HTTP proxy. Following the example in #1884 , I assume hyper::upgrade::on() should be used on the response to the CONNECT request, but it returns an error "upgrade expected but low level API in use". 
I see another crate "hyper-proxy" implemented its own logic to check for response "HTTP/1.1 200", so I wonder if client side HTTP CONNECT is still supported.

**Comment by seanmonstar** [comment, marked as answer]
It is still supported, you need to call `Connection::with_upgrades()` to enable them. The error message could probably be adjusted.
